Tests/AdyenTests/Adyen Tests/UI/Form/Items/Text/FormTextItemViewTests.swift

```diff
@@ -169,20 +169,25 @@ class FormTextItemViewTests: XCTestCase {
         XCTAssertEqual(sut.titleLabel.textColor.toHexString(), sut.tintColor.toHexString())
     }
 
-    func testSanitization() throws {
+    func testTextFieldSanitizationGivenNonAllowedCharactersShouldSanitizeAndFormatInput() throws {
         // Given
-        let cardValidator = CardNumberValidator(isLuhnCheckEnabled: false, isEnteredBrandSupported: false)
-        let cardFormatter = CardNumberFormatter()
-        item.validator = cardValidator
-        item.formatter =  cardFormatter
-
+        let validator = CardNumberValidator(isLuhnCheckEnabled: false, isEnteredBrandSupported: false)
+        let formatter = CardNumberFormatter()
+        item.validator = validator
+        item.formatter =  formatter
         sut = FormTextItemView(item: item)
 
-        sut.textField.text = "541asdfasdf"
+        let expectedItemValue = "22224000"
+        let expectedFormattedValue = formatter.formattedValue(for: expectedItemValue)
+
+        // When
+        sut.textField.text = "2222XF%4000"
         sut.textField.sendActions(for: .editingChanged)
-        wait(for: .seconds(1))
 
-        XCTAssertEqual(sut.textField.text, "541")
+        // Then
+        XCTAssertEqual(expectedItemValue, item.value)
```

Comment:
that's actually not correct, the `item.value` should be the sanitized value, but not formatted.

Reply:
Here `expectedItemValue` is the sanitized value, there's no formatting in "22224000"

Reply:
yeah, its coincidence that sanitized is actually equal to formatted, ok, no problem then.